Add normative language about expression nesting (#555)
unicode-org/message-format-wg: spec/syntax.md

> > This placeholder contains an {|expression with a literal|}
> > This placeholder references a {$variable}
> > This placeholder references a function on a variable: {$variable :function with=options}
> > This placeholder has an option with a variable: {:function option=$variable}

Reviewer: I think the first two examples here are overly cute.
```suggestion
> This placeholder contains a literal expression: {|literal|}
> This placeholder contains a variable expression: {$variable}
> This placeholder references a function on a variable: {$variable :function with=options}
> This placeholder contains a function expression with a variable-valued option: {:function option=$variable}
```

Author: ... but I _like_ cute examples. 🐱  🐈 😺 

Oh, all right... no fun in the specification...

> Hello, {$userObj :person firstName=long}!
> ```

> > A _message_ with a _variable_ as an option:

Reviewer: ```suggestion
> A _message_ using the `:list` _function_ with a _variable_-valued `maxEntries` option:
```

Author: Fixed with slightly different wording
